### PR TITLE
fix!: remove ability to specify ingress paths (#138)

### PR DESCRIPTION
**Pull request by khvn26**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have bumped the version number in `/charts/flagsmith/Chart.yaml` in the section `version` or I'm merging to a
      release branch

## Changes

This PR removes the ability to set custom paths for ingresses (issue #138). 

## How did you test this code?

Deployed it on a local cluster with ingresses enabled.

### CODE REVIEW

File: charts/flagsmith/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: flagsmith
 description: Flagsmith
 type: application
-version: 0.18.1
+version: 0.19.0
```
[the remark this answers is not in the document]
Comment (khvn26) [reply charts/flagsmith/Chart.yaml:5]: Decided on keeping the pre-1.0 versioning.

**Review by matthewelwell — APPROVED** [review]

**Comment by nishant-shah-social** [comment]
@khvn26 I am not 100% sure but mostly looks like I am facing an issue because of this change. Following is the error I am getting in the helm config while trying to deploy the workloads
```
Ingress.extensions "flagsmith-frontend" is invalid: spec.rules[0].host: Invalid value: "map[host:flagsmith.mydomain paths:[/]]": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```
I tried changing the template ingress-frontend.yaml from
```
  - host: {{ . | quote }}
  ```
to
```
  - host: {{ .host | quote }}
 ```
and it worked fine

**Comment by khvn26** [comment]
Hello @nishant-shah-social! This is a breaking change. To make it work, you should change your values YAML file to 

```
    hosts:
      - your-configured-hostname
```

Sorry for not communicating this properly in our release notes. We'll be sure to document such changes in the future, should we introduce them. We will consider releasing a patch to add backwards compatibility. Do keep in mind, however, that these path specifications do not work and in actuality did not work before. See https://github.com/Flagsmith/flagsmith-charts/issues/138 for details.

**Comment by ncalibey** [comment]
So my understanding of #138 is that the issue had to do with the path specification being ignored on the front-end code, but there was no indication that the same was true for the API server. Previously we were using NGINX rewrite rules for our path, but when we tried to upgrade to `v0.20.0`, we got the following:

> Error: release flagsmith failed, and has been uninstalled due to atomic being set: Ingress.extensions "flagsmith-api" is invalid: spec.rules[0].host: Invalid value: "map[host:<nil> paths:[/flagsmith/(.*)]]": a wildcard DNS-1123 subdomain must start with '*.', followed by a valid DNS subdomain, which must consist of lower case alphanumeric characters, '-' or '.' and end with an alphanumeric character (e.g. '*.example.com', regex used for validation is '\*\.[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')

As you can see, we'd use `/flagsmith/(.*)` as part of our rewrite, but that no longer works. We can still deploy on `v0.18.0`, which leads me to think it was this change that broke it. Is there a reason I'm missing that the API ingress was changed as well?

**Comment by dabeeeenster** [comment]
Hi @ncalibey if there's additional issues can you raise a new Github issue as this PR is merged

**Pull request merged**